src/components/Map.vue

```diff
@@ -110,21 +111,23 @@ export default createComponent({
         }
 
         if (feature) {
-          /*
-          if ('setStyle' in feature) {
-            const zoomedStyle = (feature.get('defaultStyle') as Style).clone()
-            const originalImage = zoomedStyle.getImage().clone()
-            originalImage.setScale(1.25)
-            zoomedStyle.setImage(originalImage)
-            zoomedStyle.setZIndex(2)
-            feature.setStyle(zoomedStyle)
-          }
-          */
           event('clickFactoryPin')
-          openFactoryDetail(feature)
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          if ((context.root as any).$vuetify.breakpoint.mdAndUp) {
-            expandFactoryDetail()
+          if (feature.get('factoryId')) {
+            openFactoryDetail(feature)
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            if ((context.root as any).$vuetify.breakpoint.mdAndUp) {
+              expandFactoryDetail()
+            }
+          } else {
+            // eslint-disable-next-line
+            // @ts-ignore
+            const p: Point = feature.getGeometry()
+            // eslint-disable-next-line
+            // @ts-ignore
+            const c: Coordinate = p.getCoordinates()
+            mapControllerRef?.value?.mapInstance?.map?.getView().setCenter(c)
+            const zoom = mapControllerRef?.value?.mapInstance?.map?.getView().getZoom()
+            mapControllerRef?.value?.mapInstance?.map?.getView().setZoom(zoom || 0 + 1)
```

Comment:
Should be this?

```suggestion
            mapControllerRef?.value?.mapInstance?.map?.getView().setZoom((zoom || 0) + 1)
```

Otherwise the zooming seems not working on the preview app